### PR TITLE
✨ Destination DuckDB: Use latest DuckDB 0.9.1 libraries 🚨(breaking change)🚨

### PR DESCRIPTION
**Pull request by aaronsteers**
## What

Use the latest DuckDB version as of today, which is `v0.9.1`.

This re-enables connectivity to MotherDuck, which now uses DuckDB `0.9.x`.

## How

Simple bump of the DuckDB libraries. No code changes.

## Recommended reading order
1. First read the contributing docs, which document this process.
2. (Everything else.)

## 🚨 User Impact 🚨

The `v0.9.x` DuckDB database format is not backwards compatible with versions prior to `0.9.0`.

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`556c4ae10d`](https://github.com/airbytehq/airbyte/pull/31744/commits/556c4ae10d470025998bd378b44c4f30f2334d0f)) - ❌

⏲️  Total pipeline duration: 33.79s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build destination-duckdb docker image for platform(s) linux/amd64 | ❌       |
| Code format checks                                                | ✅       |
| Validate metadata for destination-duckdb                          | ❌       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ❌       |
| QA checks                                                         | ❌       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/destination-duckdb_bump-to-duckdb-091/1698091339/556c4ae10d470025998bd378b44c4f30f2334d0f/destination-duckdb/0.2.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:556c4ae10d470025998bd378b44c4f30f2334d0f)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

File: airbyte-integrations/connectors/destination-duckdb/metadata.yaml [hunk 1]
```diff
@@ -14,11 +14,18 @@ data:
...
+        upgradeDeadline: "2023-12-31"
   documentationUrl: https://docs.airbyte.com/integrations/destinations/duckdb
   tags:
     - language:python
   ab_internal:
     sl: 100
     ql: 100
   supportLevel: community
+  breaking_changes:
+    - 0.3.0
```
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-duckdb/metadata.yaml:30]:
```suggestion
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`a382ec1fca`](https://github.com/airbytehq/airbyte/pull/31744/commits/a382ec1fca94b7f9d3755320bd9d066130065a66)) - ❌

⏲️  Total pipeline duration: 20.74s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build destination-duckdb docker image for platform(s) linux/amd64 | ❌       |
| Code format checks                                                | ✅       |
| Validate metadata for destination-duckdb                          | ✅       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ✅       |
| QA checks                                                         | ❌       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/destination-duckdb_bump-to-duckdb-091/1698092071/a382ec1fca94b7f9d3755320bd9d066130065a66/destination-duckdb/0.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:a382ec1fca94b7f9d3755320bd9d066130065a66)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`0ce82e4fa3`](https://github.com/airbytehq/airbyte/pull/31744/commits/0ce82e4fa340a47b329c29243ff3b657f4b9dc64)) - ❌

⏲️  Total pipeline duration: 21.80s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build destination-duckdb docker image for platform(s) linux/amd64 | ❌       |
| Code format checks                                                | ✅       |
| Validate metadata for destination-duckdb                          | ✅       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ✅       |
| QA checks                                                         | ❌       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/destination-duckdb_bump-to-duckdb-091/1698109433/0ce82e4fa340a47b329c29243ff3b657f4b9dc64/destination-duckdb/0.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:0ce82e4fa340a47b329c29243ff3b657f4b9dc64)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

**Comment by aaronsteers** [comment]
@alafanechere, @bnchrch - I've copied over the base image config from:
- https://github.com/airbytehq/airbyte/pull/31543

Lmk if this looks okay.

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`18ea47b056`](https://github.com/airbytehq/airbyte/pull/31744/commits/18ea47b0567a3ff2479728f975730d41d6e1ef30)) - ❌

⏲️  Total pipeline duration: 01mn32s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build destination-duckdb docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                        | ✅       |
| Integration tests                                                 | ✅       |
| Acceptance tests                                                  | ✅       |
| Code format checks                                                | ✅       |
| Validate metadata for destination-duckdb                          | ✅       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ✅       |
| QA checks                                                         | ❌       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/destination-duckdb_bump-to-duckdb-091/1698109990/18ea47b0567a3ff2479728f975730d41d6e1ef30/destination-duckdb/0.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:18ea47b0567a3ff2479728f975730d41d6e1ef30)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

**Review by alafanechere — APPROVED** [review]
The migration to the base image looks good to me! Thank you.
Please write a migration guide in `docs/integrations/destinations/duckdb-migrations.md` to document how users should upgrade DuckDb.

File: airbyte-integrations/connectors/destination-duckdb/metadata.yaml [hunk 2]
```diff
@@ -1,8 +1,10 @@
 data:
+  connectorBuildOptions:
+    baseImage: docker.io/airbyte/python-connector-base:1.1.0@sha256:bd98f6505c6764b1b5f99d3aedc23dfc9e9af631a62533f60eb32b1d3dbab20c
   connectorSubtype: database
   connectorType: destination
   definitionId: 94bd199c-2ff0-4aa2-b98e-17f0acb72610
-  dockerImageTag: 0.2.1
+  dockerImageTag: 0.3.0
```
Comment (alafanechere) [inline airbyte-integrations/connectors/destination-duckdb/metadata.yaml:7]:
```suggestion
  dockerImageTag: 1.0.0
```

It's a breaking change, can we bump be a major version?

File: airbyte-integrations/connectors/destination-duckdb/metadata.yaml [hunk 3]
```diff
@@ -14,6 +16,11 @@ data:
     oss:
       enabled: true
   releaseStage: alpha
+  releases:
+    breakingChanges:
+      0.3.0:
```
Comment (bnchrch) [inline airbyte-integrations/connectors/destination-duckdb/metadata.yaml:21]:
```suggestion
      1.0.0:
```
Just adding the major version suggestion here as well

**Review by bnchrch — DISMISSED** [review]
Looks like the only issue here is making it a major version change instead of a minor

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`8b601e9fde`](https://github.com/airbytehq/airbyte/pull/31744/commits/8b601e9fdeb15455a9f17e0872dc2f5f190da942)) - ❌

⏲️  Total pipeline duration: 01mn41s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build destination-duckdb docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                        | ✅       |
| Integration tests                                                 | ✅       |
| Acceptance tests                                                  | ✅       |
| Code format checks                                                | ✅       |
| Validate metadata for destination-duckdb                          | ✅       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ✅       |
| QA checks                                                         | ❌       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/destination-duckdb_bump-to-duckdb-091/1698165248/8b601e9fdeb15455a9f17e0872dc2f5f190da942/destination-duckdb/0.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:8b601e9fdeb15455a9f17e0872dc2f5f190da942)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`3d7d6b6fb0`](https://github.com/airbytehq/airbyte/pull/31744/commits/3d7d6b6fb03ab1917a6fc9d951a7ea3a776be74c)) - ✅

⏲️  Total pipeline duration: 01mn15s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build destination-duckdb docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                        | ✅       |
| Integration tests                                                 | ✅       |
| Acceptance tests                                                  | ✅       |
| Code format checks                                                | ✅       |
| Validate metadata for destination-duckdb                          | ✅       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ✅       |
| QA checks                                                         | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/destination-duckdb_bump-to-duckdb-091/1698165713/3d7d6b6fb03ab1917a6fc9d951a7ea3a776be74c/destination-duckdb/0.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:3d7d6b6fb03ab1917a6fc9d951a7ea3a776be74c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`f1697c11d1`](https://github.com/airbytehq/airbyte/pull/31744/commits/f1697c11d103ee7b4cca9e554905766b7c38cb1e)) - ✅

⏲️  Total pipeline duration: 50.28s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build destination-duckdb docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                        | ✅       |
| Integration tests                                                 | ✅       |
| Acceptance tests                                                  | ✅       |
| Code format checks                                                | ✅       |
| Validate metadata for destination-duckdb                          | ✅       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ✅       |
| QA checks                                                         | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/destination-duckdb_bump-to-duckdb-091/1698166241/f1697c11d103ee7b4cca9e554905766b7c38cb1e/destination-duckdb/0.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:f1697c11d103ee7b4cca9e554905766b7c38cb1e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`889bdcbef3`](https://github.com/airbytehq/airbyte/pull/31744/commits/889bdcbef30b63ed24f5b15e6054c063a1d1c718)) - ✅

⏲️  Total pipeline duration: 53.16s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build destination-duckdb docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                        | ✅       |
| Integration tests                                                 | ✅       |
| Acceptance tests                                                  | ✅       |
| Code format checks                                                | ✅       |
| Validate metadata for destination-duckdb                          | ✅       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ✅       |
| QA checks                                                         | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/destination-duckdb_bump-to-duckdb-091/1698179558/889bdcbef30b63ed24f5b15e6054c063a1d1c718/destination-duckdb/0.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:889bdcbef30b63ed24f5b15e6054c063a1d1c718)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`eec1bc82ed`](https://github.com/airbytehq/airbyte/pull/31744/commits/eec1bc82ede28b3ccfe9303d37c4961b525aa1c8)) - ✅

⏲️  Total pipeline duration: 44.46s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build destination-duckdb docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                        | ✅       |
| Integration tests                                                 | ✅       |
| Acceptance tests                                                  | ✅       |
| Code format checks                                                | ✅       |
| Validate metadata for destination-duckdb                          | ✅       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ✅       |
| QA checks                                                         | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/destination-duckdb_bump-to-duckdb-091/1698179948/eec1bc82ede28b3ccfe9303d37c4961b525aa1c8/destination-duckdb/0.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:eec1bc82ede28b3ccfe9303d37c4961b525aa1c8)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`e19842a6a4`](https://github.com/airbytehq/airbyte/pull/31744/commits/e19842a6a4bd61da84257cfa0bc566e1df2f9172)) - ✅

⏲️  Total pipeline duration: 32.52s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build destination-duckdb docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                        | ✅       |
| Integration tests                                                 | ✅       |
| Acceptance tests                                                  | ✅       |
| Code format checks                                                | ✅       |
| Validate metadata for destination-duckdb                          | ✅       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ✅       |
| QA checks                                                         | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/destination-duckdb_bump-to-duckdb-091/1698180251/e19842a6a4bd61da84257cfa0bc566e1df2f9172/destination-duckdb/0.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e19842a6a4bd61da84257cfa0bc566e1df2f9172)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`2289de6b30`](https://github.com/airbytehq/airbyte/pull/31744/commits/2289de6b30ec68e4eb9a21cd6727897864773c0d)) - ✅

⏲️  Total pipeline duration: 28.47s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build destination-duckdb docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                        | ✅       |
| Integration tests                                                 | ✅       |
| Acceptance tests                                                  | ✅       |
| Code format checks                                                | ✅       |
| Validate metadata for destination-duckdb                          | ✅       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ✅       |
| QA checks                                                         | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/destination-duckdb_bump-to-duckdb-091/1698182853/2289de6b30ec68e4eb9a21cd6727897864773c0d/destination-duckdb/0.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:2289de6b30ec68e4eb9a21cd6727897864773c0d)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

File: airbyte-integrations/connectors/destination-duckdb/metadata.yaml (hunk 2 above)
Comment (aaronsteers) [reply airbyte-integrations/connectors/destination-duckdb/metadata.yaml:7]: We spoke offline. This is `community` connector and doesn't receive official support.

Also, DuckDB's own database version files are incompatible across minor versions, and the best way to respect this volatility is to keep pre-1.0 connector status until we have a handle on stability overall.

Last - there's a desire to rewrite this in Java before we give official/certified status to the connector.

File: airbyte-integrations/connectors/destination-duckdb/metadata.yaml (hunk 3 above)
Comment (aaronsteers) [reply airbyte-integrations/connectors/destination-duckdb/metadata.yaml:21]: Hi, @bnchrch 

https://github.com/airbytehq/airbyte/pull/31744/files#r1370836343

From my response on Augustine's comment above:

> We spoke offline. This is community connector and doesn't receive official support.
>
> Also, DuckDB's own database version files are incompatible across minor versions, and the best way to respect this volatility is to keep pre-1.0 connector status until we have a handle on stability overall.
>
> Last - there's a desire to rewrite this in Java before we give official/certified status to the connector.

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`d5c6acacfa`](https://github.com/airbytehq/airbyte/pull/31744/commits/d5c6acacfa5851b32e22ecdacd7b2bc8605be59a)) - ✅

⏲️  Total pipeline duration: 36.96s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Build destination-duckdb docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                        | ✅       |
| Integration tests                                                 | ✅       |
| Acceptance tests                                                  | ✅       |
| Code format checks                                                | ✅       |
| Validate metadata for destination-duckdb                          | ✅       |
| Connector version semver check                                    | ✅       |
| Connector version increment check                                 | ✅       |
| QA checks                                                         | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/destination-duckdb_bump-to-duckdb-091/1698184024/d5c6acacfa5851b32e22ecdacd7b2bc8605be59a/destination-duckdb/0.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:d5c6acacfa5851b32e22ecdacd7b2bc8605be59a)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

File: airbyte-integrations/connectors/destination-duckdb/metadata.yaml (hunk 3 above)
Comment (bnchrch) [reply airbyte-integrations/connectors/destination-duckdb/metadata.yaml:21]: Ah fair enough @aaronsteers !

I was going to dig my heels in on "no major change no breaking change" but after a discussion with @pedroslopez he pointed out weve been doing this same pattern of minor version changes can be breaking if the connector is v0

langchain is the example that came to mind
https://github.com/airbytehq/airbyte/blob/c92837c8d7e7f5f84f63547f3f98ed541fb4defb/airbyte-integrations/connectors/destination-langchain/metadata.yaml

and that seems to line up with the official interpretation of semver when the version is still 0.

**TLDR: LGTM!**
